app/src/fdroid/java/org/fossasia/openevent/general/search/GeoLocationViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package org.fossasia.openevent.general.search
+
+import android.app.Activity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class GeoLocationViewModel : ViewModel() {
+    private val mutableLocation = MutableLiveData<String>()
+    val location: LiveData<String> = mutableLocation
+    private val mutableVisibility = MutableLiveData<Boolean>()
+    val currentLocationVisibility: LiveData<Boolean> = mutableVisibility
+
+    fun getVisibility() {
+        mutableVisibility.value = false
+    }
```

Comment:
No need of get visibility. Just do this in configure

Reply:
But location request is from the fragment. Configure method is required location permission if I call it at the start. And location permission cannot be called at the start because it is not required for fdroid. So, I call configure conditionally.

Reply:
OK, so just call this in onStart

Reply:
Updated

Reply:
Still same

Reply:
I have called it from on start of the fragment. I am sorry to not getting it.